src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4008,8 +4008,10 @@ private BoundExpression MakeBadExpressionForObjectCreation(ObjectCreationExpress
                 expanded,
                 argsToParamsOpt,
                 resultKind,
-                boundMember.Type,
-                hasErrors);
+                binderOpt: this,
+                useSetterForDefaultArgumentGeneration: true,
```

Comment:
This code path is while binding left of object member initializer, so it should always invoke the setter.